### PR TITLE
feat: add sequence command for Mermaid sequence diagrams

### PR DESCRIPTION
**Pull request by carlos-alm**
## Summary

- Adds `codegraph sequence <name>` command that generates Mermaid `sequenceDiagram` output from the call graph
- Participants are files (not individual functions), keeping diagrams readable and matching Greptile's style
- BFS forward from entry point with optional `--dataflow` flag to annotate parameter names and return arrows
- Exposed via CLI, MCP tool, and programmatic API

### Example: `codegraph sequence flowData -T --depth 2`

```mermaid
sequenceDiagram
    participant flow as flow.js
    participant db as db.js
    participant paginate as paginate.js
    participant queries as queries.js
    flow->>db: openReadonlyOrFail
    flow->>flow: findBestMatch
    flow->>flow: entryPointType
    flow->>queries: isTestFile
    flow->>paginate: paginateResult
    db->>db: findDbPath
    flow->>queries: isTestFile
    paginate->>paginate: paginate
    note right of flow: Truncated at depth 2
```

## Files changed

| File | Change |
|------|--------|
| `src/sequence.js` | **New** — core module: `sequenceData()`, `sequenceToMermaid()`, `sequence()` |
| `tests/integration/sequence.test.js` | **New** — 11 integration tests with hand-crafted DB |
| `src/cli.js` | Register `sequence <name>` command |
| `src/mcp.js` | Add `sequence` MCP tool definition + handler |
| `src/index.js` | Export programmatic API |
| `tests/unit/mcp.test.js` | Add `sequence` to expected tool names |
| `CLAUDE.md` | Architecture table + command reference |

## Deferred (follow-up PR)

- Add `sequence` to `BATCH_COMMANDS` in `src/batch.js`
- Add `sequence: 50` to `MCP_DEFAULTS` in `src/paginate.js`

## Test plan

- [x] All 1443 tests pass (79 test files)
- [x] Biome lint clean on all changed files
- [x] `codegraph sequence buildGraph -T` produces valid Mermaid (15 participants, 218 messages)
- [x] `codegraph sequence buildGraph -T -j` outputs valid JSON
- [x] `codegraph sequence flowData -T --depth 2` produces compact diagram

### CODE REVIEW

**Comment by carlos-alm** [comment]
## Sequence diagram (generated by this PR)

`codegraph sequence buildGraph -T --depth 3`

```mermaid
sequenceDiagram
    participant builder as builder.js
    participant ast as ast.js
    participant cfg as cfg.js
    participant complexity as complexity.js
    participant config as config.js
    participant constants as constants.js
    participant dataflow as dataflow.js
    participant db as db.js
    participant journal as journal.js
    participant logger as logger.js
    participant native as native.js
    participant parser as parser.js
    participant registry as registry.js
    participant resolve as resolve.js
    participant structure as structure.js
    builder->>db: openDb
    builder->>db: initSchema
    builder->>config: loadConfig
    builder->>parser: getActiveEngine
    builder->>logger: info
    builder->>db: getBuildMeta
    builder->>builder: loadPathAliases
    builder->>constants: normalizePath
    builder->>builder: collectFiles
    builder->>builder: getChangedFiles
    builder->>parser: parseFilesAuto
    builder->>cfg: buildCFGData
    builder->>dataflow: buildDataflowEdges
    builder->>db: closeDb
    builder->>journal: writeJournalHeader
    builder->>logger: debug
    builder->>builder: purgeFilesFromGraph
    builder->>builder: readFileSafe
    builder->>builder: fileStat
    builder->>builder: fileHash
    builder->>resolve: resolveImportsBatch
    builder->>builder: getResolved
    builder->>builder: isBarrelFile
    builder->>builder: resolveBarrelExport
    builder->>resolve: computeConfidence
    builder->>structure: buildStructure
    builder->>structure: classifyNodeRoles
    builder->>ast: buildAstNodes
    builder->>complexity: buildComplexityMetrics
    builder->>parser: ensureWasmTrees
    builder->>logger: warn
    builder->>db: setBuildMeta
    builder->>registry: registerRepo
    db->>db: acquireAdvisoryLock
    db->>logger: debug
    config->>logger: debug
    config->>config: mergeConfig
    config->>config: resolveSecrets
    config->>config: applyEnvOverrides
    parser->>parser: resolveEngine
    cfg->>parser: createParsers
    cfg->>complexity: findFunctionNode
    cfg->>cfg: buildFunctionCFG
    cfg->>logger: info
    dataflow->>parser: createParsers
    dataflow->>dataflow: extractDataflow
    dataflow->>dataflow: resolveNode
    dataflow->>logger: info
    db->>db: releaseAdvisoryLock
    journal->>logger: warn
    resolve->>native: loadNative
    resolve->>resolve: convertAliasesForNative
    resolve->>constants: normalizePath
    structure->>constants: normalizePath
    structure->>logger: debug
    structure->>structure: median
    ast->>logger: debug
    ast->>ast: findParentDef
    ast->>ast: walkAst
    complexity->>parser: createParsers
    complexity->>parser: getParser
    complexity->>complexity: findFunctionNode
    complexity->>complexity: computeAllMetrics
    complexity->>logger: info
    parser->>parser: createParsers
    registry->>registry: loadRegistry
    registry->>registry: saveRegistry
    registry->>logger: debug
    note right of builder: Truncated at depth 3
```

@greptileai

File: src/sequence.js [hunk 1]
```diff
@@ -0,0 +1,435 @@
...
+  for (const msg of seqResult.messages) {
+    const arrow = msg.type === 'return' ? '-->>' : '->>';
+    lines.push(`    ${msg.from}${arrow}${msg.to}: ${escapeMermaid(msg.label)}`);
+  }
+
+  if (seqResult.truncated) {
+    lines.push(
+      `    note right of ${seqResult.participants[0]?.id}: Truncated at depth ${seqResult.depth}`,
+    );
+  }
```
[the remark this answers is not in the document]
Comment (carlos-alm) [reply src/sequence.js]: Fixed — guarded the truncation note on `seqResult.participants.length > 0` and removed the unnecessary optional chaining. Also added a dedicated test (`no truncation note when participants empty`) that asserts no `note right of` or `undefined` appears in the Mermaid output when participants is empty but truncated is true.

File: src/sequence.js [hunk 2]
```diff
@@ -0,0 +1,435 @@
...
+ * become self-messages. This keeps diagrams readable and matches typical
+ * sequence-diagram conventions.
+ */
+
+import { openReadonlyOrFail } from './db.js';
+import { paginateResult, printNdjson } from './paginate.js';
+import { isTestFile, kindIcon } from './queries.js';
+import { FRAMEWORK_ENTRY_PREFIXES } from './structure.js';
+
+// ─── findBestMatch (copied from flow.js — same pattern) ─────────────
```
[the remark this answers is not in the document]
Comment (carlos-alm) [reply src/sequence.js]: Fixed — removed the local `findBestMatch` copy from both `sequence.js` and `flow.js`. Exported the canonical `findMatchingNodes` from `queries.js` and replaced all calls with `findMatchingNodes(db, name, opts)[0] ?? null`. Single source of truth now.

File: tests/integration/sequence.test.js [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+import { afterAll, beforeAll, describe, expect, test } from 'vitest';
+import { initSchema } from '../../src/db.js';
+import { sequenceData, sequenceToMermaid } from '../../src/sequence.js';
+
+// ─── Helpers ───────────────────────────────────────────────────────────
+
+function insertNode(db, name, kind, file, line) {
+  return db
+    .prepare('INSERT INTO nodes (name, kind, file, line) VALUES (?, ?, ?, ?)')
+    .run(name, kind, file, line).lastInsertRowid;
```
[the remark this answers is not in the document]
Comment (carlos-alm) [reply tests/integration/sequence.test.js]: Fixed — added a \ block with a dedicated fixture DB that inserts \ rows (\ + \). Four tests cover: return arrows appear with correct labels, call labels annotated with parameter names, no annotations when \, and Mermaid dashed \ arrows for returns.
